server/src/main/java/org/elasticsearch/index/mapper/InferenceMetadataFieldsMapper.java

```diff
@@ -26,7 +28,37 @@
  * the field name for removal from _source.
  */
 public abstract class InferenceMetadataFieldsMapper extends MetadataFieldMapper {
-    public static final FeatureFlag INFERENCE_METADATA_FIELDS_FEATURE_FLAG = new FeatureFlag("inference_metadata_fields");
```

Comment:
Nitpick - there's no reason not to keep the FF in as well, in case we can remove this system property sooner?

Reply:
We can add the feature flag back with a one line change if/when we want to re-introduce it. In the meantime, 
IMO better to not have a feature flag we know is not being used.